### ISSUE TITLE
Coq lets me close a module in the middle of a proof

### ISSUE THREAD

**Issue by cpitclaudel** [coq/coq#14001]
#### Description of the problem

This is weird:

```coq
Module test.
  Lemma t: True.
End test.
exact I.
Qed.
```

This is broken:

```
Module test2.
  Lemma t': True.
End test2.
Fail destruct true.
(* Error: Cannot define ?t with term "let b := true in ?t" that would depend on itself. *)
exact I.
Qed.
```

And this is an anomaly:

```
Module test3.
  Lemma t': 0 = 0.
End test3.
Fail destruct true.
(*
Error: Anomaly "Signature and its instance do not match."
Please report at http://coq.inria.fr/bugs/.
 *)
```

#### Coq Version

8.13.1

**Comment by ejgallego** [comment]
Fixed in #13790 , it was not backported to 8.13 see PR for reasoning.

**Comment by cpitclaudel** [comment]
Thanks.  I need to find a better way to search the open bug DB.
